core/unit_test/TestAtomics.hpp

```diff
@@ -308,32 +265,70 @@ T CASLoopSerial(int loop) {
 }
 
 //----------------------------------------------
-//--------------atomic_exchange-----------------
+//--------------atomic_compare_exchange_strong--
 //----------------------------------------------
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+#ifdef KOKKOS_ENABLE_DEPRECATION_WARNINGS
+KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()
+#endif
 template <class T, class DEVICE_TYPE>
-struct ExchFunctor {
+struct DeprecatedCASFunctor {
   using execution_space = DEVICE_TYPE;
   using type            = Kokkos::View<T, execution_space>;
 
-  type data, data2;
+  type data;
 
   KOKKOS_INLINE_FUNCTION
-  void operator()(int i) const {
-    T old = Kokkos::atomic_exchange(&data(), (T)i);
-    Kokkos::atomic_fetch_add(&data2(), old);
+  void operator()(int) const {
+    T newval, assumed;
+
+    do {
+      assumed = Kokkos::volatile_load(&data());
```

Comment:
@crtrott do you have a better idea?

Reply:
Agreed `atomic_load` is probably better, however we could also see that as an integral part of testing volatile_load and deprecating it?

Reply:
Yeah I guess I am ok with the volatile-load